syncd/syncd_flex_counter.cpp

```diff
@@ -522,9 +543,36 @@ void FlexCounter::flexCounterThread(void)
     while (m_runFlexCounterThread)
     {
         {
-            std::lock_guard<std::mutex> lock(g_mutex);
-            collectCounters(countersTable);
-            runPlugins(db);
+            std::map<std::string, std::vector<swss::FieldValueTuple>> portStats;
+            std::map<std::string, std::vector<swss::FieldValueTuple>> queueStats;
+            std::map<std::string, std::vector<swss::FieldValueTuple>> queueAttrs;
+            collectCounters(portStats, queueStats, queueAttrs);
```

Comment:
collectCounters has to be guarded by lock because it uses structures that can be updated from another thread

Reply:
I am not sure there is some structures shared in collectCounters among threads. It's only querying sai attribute values here.

Reply:
I am not sure there is some structure shared among threads within collectCounters; it's just querying sai attribute values.